students/EricAdams/session03/mailroom.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+
+# Data
+donors_and_donations = ['Steve Jobs', [5, 10, 15], 'Bill Gates', [
+    30, 40, 50], 'Barak Obama', [50], 'Donald Trump', [
+    1, 2, 3], 'Melinda Gates', [10, 20, 30]]
+#
+# “Send a Thank You”, “Create a Report” or “quit”
+#
+
+
+def create_a_report():
+    """ Print out a report of donors, total given number of donations
+     and average donation
+    """
+    donor_name = []
+    donor_total_given = []
+    donor_total_num_of_gifts = []
+    total_given = 0.0
+    average_gift = []
+    list_of_donations = []
+    average_of_gifts = 0
+    print('{:<20}|{:^15}|{:^15}|{:^15}'.format(
+        'Donor Name', 'Total Given', 'Num Gifts', 'Average Gift'))
+    print('-' * 65)
+    # Separate the individual donors and donations into separate lists
+    i = 0
+    while i < len(donors_and_donations):
+        donor_name.append(donors_and_donations[i])
+        list_of_donations.append(donors_and_donations[i + 1])
```

Comment:
isn't this getting awkward? better to have them grouped in the first place:

```
donors_and_donations = [ ('Steve Jobs', [5, 10, 15]),
                         ('Bill Gates', [34, 54]),
                          ...
```
you can nest tuples and lists arbitrarily deep...